### PR TITLE
Fix : https://github.com/twilio-labs/vscode-twilio/issues/13

### PR DESCRIPTION
**Pull request by gagandeepp**
This pull request will fix issue 13
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,6 +94,9 @@ Moves an active deployment from one environment to another.
 ##### `Twilio Serverless: Deploy`
 Deploy functions from your local to the specified environment.
 
+##### `Twilio Serverless: List`
+List functions from your local to the specified environment.
```
[the remark this answers is not in the document]
Comment (gagandeepp) [reply README.md]: sounds good. Fixed

File: README.md [hunk 2]
```diff
@@ -94,6 +94,9 @@ Moves an active deployment from one environment to another.
 ##### `Twilio Serverless: Deploy`
 Deploy functions from your local to the specified environment.
 
+##### `Twilio Serverless: List`
```
[the remark this answers is not in the document]
Comment (gagandeepp) [reply README.md]: sounds good. Fixed

File: package.json [hunk 3]
```diff
@@ -50,6 +51,10 @@
 			{
 				"command": "extension.createEnvironment",
 				"title": "Twilio Serverless: Create Environment"
+			},
+			{
+				"command": "extension.listFunction",
+				"title": "Twilio Serverless: List functions"
```
[the remark this answers is not in the document]
Comment (gagandeepp) [reply package.json]: sounds good. Fixed

File: extension.js [hunk 4]
```diff
@@ -22,14 +23,16 @@ function activate(context) {
 	const deployFunction = vscode.commands.registerCommand('extension.deploy', deploy);
  	const activateFunction = vscode.commands.registerCommand('extension.activate', activateFn);
 	const createEnvironment = vscode.commands.registerCommand('extension.createEnvironment', createEnv);
+	const listFunction = vscode.commands.registerCommand('extension.list',list);
```
[the remark this answers is not in the document]
Comment (gagandeepp) [reply extension.js]: sounds good.Fixed

File: package.json [hunk 5]
```diff
@@ -22,7 +22,8 @@
 		"onCommand:extension.start",
 		"onCommand:extension.deploy",
 		"onCommand:extension.activate",
-		"onCommand:extension.createEnvironment"
+		"onCommand:extension.createEnvironment",
+		"onCommand:extension.list"
```
[the remark this answers is not in the document]
Comment (gagandeepp) [reply package.json]: sounds good Fxed

File: extension.js [hunk 6]
```diff
@@ -22,14 +23,16 @@ function activate(context) {
...
 
 	context.subscriptions.push(
 		activateFunction,
 		createEnvironment,
 		createFunction,
 		createProject,
 		deployFunction,
-		startServer
+		startServer,
+		listFunction
```
[the remark this answers is not in the document]
Comment (gagandeepp) [reply extension.js]: sounds good fixed

File: commands/list.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+const vscode = require("vscode");
+const fs = require("fs");
+const path = require('path');
+const assignTerminal = require("../helpers/assignTerminal");
+
+function list() {
```
[the remark this answers is not in the document]
Comment (gagandeepp) [reply commands/list.js]: Sounds good.Fixed

File: commands/list.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  const terminal = assignTerminal(vscode, "list");
+  terminal.show();
+  terminal.sendText(`twilio serverless:list`);
+  const wsPath = vscode.workspace.workspaceFolders[0].uri.fsPath;
+  const fnDirPath = path.join(wsPath, "functions");
+  fs.readdir(fnDirPath, (err, files) => {
```
[the remark this answers is not in the document]
Comment (gagandeepp) [reply commands/list.js]: Sounds good Fixed

File: commands/list.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+const vscode = require("vscode");
+const fs = require("fs");
+const path = require('path');
+const assignTerminal = require("../helpers/assignTerminal");
+
+function list() {
+  const terminal = assignTerminal(vscode, "list");
+  terminal.show();
+  terminal.sendText(`twilio serverless:list`);
```
[the remark this answers is not in the document]
Comment (Xabadu) [reply commands/list.js]: Hey @gagandeepp, any chance to get to this change as well? For the command you're sending to the terminal, you'll need to pass the type param so it lists the correct resources. Thanks!

File: commands/list.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+const assignTerminal = require("../helpers/assignTerminal");
+const { TwilioClientCommand } = require("@twilio/cli-core").baseCommands;
+
+const { handler, cliInfo, describe } = require("twilio-run/dist/commands/list");
+
+function list(type = 'functions') {
+  const terminal = assignTerminal(vscode, "list");
+  terminal.show();
+  terminal.sendText(`twilio serverless:list`);
+  const wsPath = vscode.workspace.workspaceFolders[0].uri.fsPath;
```
Comment (Xabadu) [inline commands/list.js]: This line shouldn't be necessary anymore

File: commands/list.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (gagandeepp) [reply commands/list.js]: Done.{lease check

File: commands/list.js (hunk 10 above)
Comment (gagandeepp) [reply commands/list.js]: Done

**Comment by gagandeepp** [comment]
@Xabadu can you add me as contributor to twilio labs repository??

**Pull request merged**